config/internal/configsource/component/configsource.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
This seems to be in the internal package, preventing downstream builds to have their own config sources. Is that by design?

Reply:
We want to experiment with this until we find the right interface.